core/container/externalbuilder/session.go

```diff
@@ -24,13 +25,18 @@ type Session struct {
 	exited     chan struct{}
 	exitErr    error
 	waitStatus syscall.WaitStatus
+	signalChan <-chan struct{}
+	doneChan   chan<- struct{}
+	WaitGroup  sync.WaitGroup
```

Comment:
This should not be exported; notice nothing else in here is exported. I'm also not convinced it's needed.